pkg/apis/extensions/v1alpha1/helper/helper.go

```diff
@@ -31,10 +31,15 @@ func ClusterAutoscalerRequired(pools []extensionsv1alpha1.WorkerPool) bool {
 	return false
 }
 
-// GetDNSRecordType returns the appropriate DNS record type (A or CNAME) for the given address.
+// GetDNSRecordType returns the appropriate DNS record type (A/AAAA or CNAME) for the given address.
 func GetDNSRecordType(address string) extensionsv1alpha1.DNSRecordType {
-	if ip := net.ParseIP(address); ip != nil && ip.To4() != nil {
-		return extensionsv1alpha1.DNSRecordTypeA
+	if ip := net.ParseIP(address); ip != nil {
+		if ip.To4() != nil {
+			return extensionsv1alpha1.DNSRecordTypeA
+		} else {
+			// Since the IP is neither nil nor IPv4 it therefore must be IPv6
+			return extensionsv1alpha1.DNSRecordTypeAAAA
+		}
```

Comment:
I don' think this comment is needed - the code is pretty readable.

```suggestion
		}
		return extensionsv1alpha1.DNSRecordTypeAAAA
```

Reply:
Sure. Just wanted to be super clear. Addressed!